relay-server/src/actors/processor.rs

```diff
@@ -1738,7 +1741,11 @@ impl EnvelopeProcessor {
                 process_value(event, &mut processor, ProcessingState::root())
                     .map_err(ProcessingError::ProcessingFailed)?;
             }
-            if let Some(config) = config.datascrubbing_settings.pii_config() {
+            let pii_config = config
+                .datascrubbing_settings
+                .pii_config()
+                .map_err(|e| ProcessingError::PiiConfigError(e.clone()))?;
```

Comment:
This will drop the event, which we should assign a dedicated discard reason to (or use the one we also use for broken project configs). Can we additionally mark the project config as invalid somehow so we reject subsequent events earlier?

Reply:
This got me thinking, the entire concept of holding a cached `Result` inside the config, and then failing every time an event needs it, seems wrong.

https://github.com/getsentry/relay/blob/8dbbc85ae5571a80010408d567d23bfa74c5efd2/relay-general/src/pii/legacy.rs#L41

I simplified the code to convert the config on serialization, thereby making this particular error a deserialization error. Because it's a bigger refactor, I put that change into its own sub-PR: https://github.com/getsentry/relay/pull/1478

Reply:
Closed the linked PR (see discussion there) and added a separate discard reason for pii conversion failure.

I would not put extra effort into invalidating the project config when this conversion fails, since having invalid data scrubbing settings is a clear bug that should be fixed on the sentry side, rather than an acceptable day-to-day case.